apps/worker/src/app/workflow/usecases/add-job/merge-or-create-digest.usecase.ts

```diff
@@ -150,9 +150,13 @@ export class MergeOrCreateDigest {
   }
 
   private getLockKey(job: JobEntity, digestKey: string | undefined, digestValue: string | number | undefined): string {
-    let resource = `environment:${job._environmentId}:template:${job._templateId}:subscriber:${job._subscriberId}`;
+    const resource = `environment:${job._environmentId}:template:${job._templateId}:subscriber:${job._subscriberId}`;
     if (digestKey && digestValue) {
-      resource = `${resource}:digestKey:${digestKey}:digestValue:${digestValue}`;
+      return `${resource}:digestKey:${digestKey}:digestValue:${digestValue}`;
+    }
+
+    if (digestValue) {
```

Comment:
What's the scenario where we have a key but not a digestValue? Is the key only the fixed `subscriberId`?

Reply:
sorry i missed this comment before merging.
we care about the following scenarios, 

v1 for valid digest we need both digestKey && digestValue. to tell you the truth i think that adding the value here will be enough as well as it should be a unique identifier and not a combination of key and value.

v2 where we have only the digestValue.

Reply:
in v2 in the end there is no digestKey only parsed key which is the digestValue, so if digestValue is missing we will use dynamic subscriberId